# Backward Elimination (Using MLxtend)
# Forward Elimination (Using MLxtend)

In [1]:
# Feature selection techniques : A feature is an attribute that has an impact on a problem or is useful for the problem, and choosing the important features for the model is known as feature selection.
# If you know the domain knowledge then in that case you use feature selection techniques.

# But now we learn how to use feature selection techniques while we don't know any prior knowledge of domain or data. But it(Backward / Forward elimination) not works every time.


# Train Test Split In Dataset
- Dataset devided into two parts
  1. train Dataset
  2. Test Dataset
     

In [4]:
import pandas as pd


In [5]:
Dataset = pd.read_csv(r"D:\Jahanvi\Data Science\ML\Boston.csv")
Dataset.head()

crim    zn  indus  chas    nox     rm   age     dis  rad  tax  ptratio  \
0  0.00632  18.0   2.31     0  0.538  6.575  65.2  4.0900    1  296     15.3   
1  0.02731   0.0   7.07     0  0.469  6.421  78.9  4.9671    2  242     17.8   
2  0.02729   0.0   7.07     0  0.469  7.185  61.1  4.9671    2  242     17.8   
3  0.03237   0.0   2.18     0  0.458  6.998  45.8  6.0622    3  222     18.7   
4  0.06905   0.0   2.18     0  0.458  7.147  54.2  6.0622    3  222     18.7   

    black  lstat  House_price  
0  396.90   4.98         24.0  
1  396.90   9.14         21.6  
2  392.83   4.03         34.7  
3  394.63   2.94         33.4  
4  396.90   5.33         36.2

In [10]:
Dataset.shape

(506, 14)

In [7]:
input_data = Dataset.iloc[:,:-1]
output_data = Dataset["House_price"]

In [8]:
from sklearn.model_selection import train_test_split

In [14]:
# test_size is optional, by default it takes test_size=0.25
x_train, x_test, y_train, y_test= train_test_split(input_data,output_data,test_size=0.25)

In [15]:
# Training input, Training output
x_train.shape, y_train.shape

((379, 13), (379,))

In [16]:
# Testing input, Testing output
x_test.shape, y_test.shape

((127, 13), (127,))

- In our dataset total 506 rows are present, we devided it into two part
- After dividing total 379 rows present for training
- After dividing total 127 rows present for testing